## DEVP-2 Project_2: Dashboard using Python ##

__Submitted  by:__

1. Ankita Singh (035011)

2. Prabhleen Kaur Pahwa (035039)

__DEPENDENCIES__

In [1]:
pip install dash plotly

Note: you may need to restart the kernel to use updated packages.


In [2]:
pip install jupyter-dash

Note: you may need to restart the kernel to use updated packages.


In [3]:
import dash_html_components as html
import dash_bootstrap_components as dbc
import pandas as pd
import plotly.express as px
from dash import dash, html, dcc, Input, Output
from jupyter_dash import JupyterDash
import seaborn as sns
import plotly.graph_objs as go
import numpy as np
import seaborn as sns
#import seaborn.objects as so
import matplotlib.pyplot as plt

C:\Users\ankit\AppData\Local\Temp\ipykernel_23016\2236185472.py:1: UserWarning: 
The dash_html_components package is deprecated. Please replace
`import dash_html_components as html` with `from dash import html`
  import dash_html_components as html


__GET DATA__

In [4]:
df = pd.read_csv(r"C:\Users\ankit\Downloads\KHOJI DATA_FINAL.csv", encoding='cp1252')
pd.DataFrame(df).head()

(Parent) ASIN (Child) ASIN         Class                         Title  \
0    B07DLYQ5TS   B07DLYQ5TS  Hanging Pots                   Diamond (5)   
1    B07DLYQ5TR   B07DLYQ5TR  Hanging Pots                Orchid (Black)   
2    B07DLWZML2   B07DLWZML2  Hanging Pots             Glory Railing (6)   
3    B083RBW381   B083RBW381  Hanging Pots          Double Hook Railing    
4    B08CY32Q45   B08CY32Q45      Planters  Rectangular Fence Window (4)   

            SKU  Sessions - Total  Sessions – Total – B2B  \
0  OJ-GDWE-6XI9             13927                      74   
1    Pots_00020              8417                      45   
2    Pots_00007              8466                      65   
3    Pots_00054              7844                      37   
4  G5-XUVN-HLWX             10251                     103   

   Session Percentage - Total  Session Percentage – Total – B2B  \
0                      0.1113                            0.0985   
1                      0.0673                            0.0599   
2                      0.0677                            0.0866   
3                      0.0627                            0.0493   
4                      0.0819                            0.1372   

   Page Views - Total  ...  Featured Offer (Buy Box) Percentage  \
0               20449  ...                               0.9929   
1               12554  ...                               0.9932   
2               13220  ...                               0.9906   
3               12220  ...                               0.9802   
4               15666  ...                               0.9951   

   Featured Offer (Buy Box) Percentage – B2B  Units Ordered  \
0                                     0.9905            585   
1                                     0.9865            445   
2                                     0.9120            379   
3                                     0.9800            330   
4                                     1.0000            349   

   Units Ordered - B2B  Unit Session Percentage  \
0                    7                   0.0420   
1                   11                   0.0529   
2                    3                   0.0448   
3                   13                   0.0421   
4                   12                   0.0340   

   Unit Session Percentage - B2B  Ordered Product Sales  \
0                              0                 231803   
1                              0                 307504   
2                              0                 222126   
3                              0                 203495   
4                              0                 278004   

   Ordered Product Sales - B2B  Total Order Items  Total Order Items - B2B  
0                         2777                556                        6  
1                         7633                397                        9  
2                         1797                379                        3  
3                         8912                298                        9  
4                         9540                291                       10  

[5 rows x 23 columns]

# BRIEF ON THE ANALYSIS

1. The data has been picked up from the *Amazon seller account of KHOJI* (a company selling virgin plastic planters). 


2. The data has 91*23 rows and coloumns, respectively and describes the page sessions, page views, orders and the total sales of each 93 unique SKUs of the company over a period of 1 year (Jan 2022- Dec-2022). 


3. __Descriptive variables__: The rows contain 93 unique SKU's of KHOJI, along with 93 unique Parent ASINs and titles. Further, the products are broadly divided into 5 classes:

      a. Hanging Pots

      b. Planters

      c. Decorative (indoor) Planters

      d. Trays

      e. Water cans 



4. __Quantitative variables__: Following are the quantitaive variables which have been further bifurcated into B2B and normal customers 

     a. *Sessions*: The number of unique visitors to your product page

     b. *Page Views*: Total number of times the product page was looked at (this can include multiple/repeat visits from the same people)

     c. *Units orders*: Total number of units sold

     d. *Ordered product sales*: Total number of units sold

     e. *Total Order Items*: Total orders that contained at least one of this SKU

     f. *Buy Box Percentage*: When a visitor landed on the product page, how often were the seller featured in the buy box (ADD TO CART)

     g. *Unit Session Percentage*: Ratio of units ordered to sessions (very similar to a conversion rate)
     

5. **Sessions total is considered a better indicator for analyzing the data as compared to Page views total because the number of unique visitors on the product page gives us a better understanding of the effectiveness of the advertisements prepared for the products.**   

__DATA PROCESSING__

In [5]:
df.columns

Index(['(Parent) ASIN', '(Child) ASIN', 'Class', 'Title', 'SKU',
       'Sessions - Total', 'Sessions – Total – B2B',
       'Session Percentage - Total', 'Session Percentage – Total – B2B',
       'Page Views - Total', 'Page Views – Total – B2B',
       'Page Views Percentage - Total', 'Page Views Percentage – Total – B2B',
       'Featured Offer (Buy Box) Percentage',
       'Featured Offer (Buy Box) Percentage – B2B', 'Units Ordered',
       'Units Ordered - B2B', 'Unit Session Percentage',
       'Unit Session Percentage - B2B', 'Ordered Product Sales',
       'Ordered Product Sales - B2B', 'Total Order Items',
       'Total Order Items - B2B'],
      dtype='object')

In [6]:
#df1 containing Normal customer data
df1 = df[['(Parent) ASIN', 
            'Class', 'Title', 
            'Sessions - Total', 
            'Page Views - Total', 
            'Units Ordered', 'Featured Offer (Buy Box) Percentage',
            'Unit Session Percentage',
            'Ordered Product Sales', 
            'Total Order Items']]

#df2 containing B2B customer data
df2=df[['(Parent) ASIN', '(Child) ASIN', 'Class', 'Title', 'SKU',
       'Sessions – Total – B2B',
       'Session Percentage – Total – B2B',
       'Page Views – Total – B2B',
       'Page Views Percentage – Total – B2B',
       'Featured Offer (Buy Box) Percentage – B2B', 
       'Units Ordered - B2B',
       'Unit Session Percentage - B2B',
       'Ordered Product Sales - B2B',
       'Total Order Items - B2B',]]

In [7]:
#Get overall total for units ordered and product sales 
def column_total(df,columns):
    return df[columns].sum(axis=0)

df1_columns = ['Units Ordered','Ordered Product Sales' ]
df1_total = column_total(df1, df1_columns)
df2_columns = ['Units Ordered - B2B','Ordered Product Sales - B2B']
df2_total = column_total(df2, df2_columns)

print('Normal Overall:',df1_total)
print('B2B Overall:',df2_total)

Normal Overall: Units Ordered               4281
Ordered Product Sales    2464035
dtype: int64
B2B Overall: Units Ordered - B2B              113
Ordered Product Sales - B2B    70838
dtype: int64


In [8]:
#Get total for units ordered and product sales for category
def get_ctg_total(df, columns, Class='Hanging Pots'):
    return df[df['Class']==Class][columns].sum()

Class = 'Hanging Pots'
df1_columns = ['Units Ordered', 'Ordered Product Sales']
df2_columns = ['Units Ordered - B2B', 'Ordered Product Sales - B2B']
ctg='Hanging Pots'
norm_total = get_ctg_total(df1, df1_columns, Class)
b2b_total = get_ctg_total(df2, df2_columns, Class)
print(f'{ctg} Total:', norm_total)
print(f'{ctg} Total (B2B):', b2b_total)

Hanging Pots Total: Units Ordered               2478
Ordered Product Sales    1313574
dtype: int64
Hanging Pots Total (B2B): Units Ordered - B2B               57
Ordered Product Sales - B2B    31188
dtype: int64


## __OBJECTIVES OF THE ANALYSIS__

__Categories' Analysis__

**a. _Normal customers_:**
        
     1. Total Sales (Dashboard)
        
     2. Total Orders (Dashboard)
        
     3. Top 2 and Bottom 2 categories based on the sessions, page view, unit ordered, ordered product sales (described using bar charts)

     4. Relation between the page view/sessions with the units ordered and the ordered products sales (Dashboard)
     
     5. Statistical Analysis

**b. _B2B customers_:**

    1. Total Sales (Dashboard)

    2. Total Orders (Dashboard)

    3. Top 2 and Bottom 2 categories based on the sessions, page view, unit ordered, ordered product sales (described using bar charts)

    4. Relation between the page view/sessions with the units ordered and the ordered products sales (Dashboard)
    
    5. Statistical Analysis


__GENERATE CARDS FOR OVERALL NUMBERS__

In [9]:
def generate_card_content(card_header, overall_value):
    card_head_style = {'textAlign':'center','fontSize':'150%'}
    card_body_style = {'textAlign':'center','fontSize':'200%'}
    card_header = dbc.CardHeader(card_header,style=card_head_style)
    card_body = dbc.CardBody(
        [
            html.H5(f"{int(overall_value):,}", className="card-title",style=card_body_style),
            html.P(
                #"Overall: {:,}".format(overall_value[0].iloc[0]),
                className="card-text",style={'textAlign':'center'}
            ),
        ]
    )
    card = [card_header,card_body]
    return card

In [14]:
def generate_cards():
    df = pd.read_csv(r"C:\Users\ankit\Downloads\KHOJI DATA_FINAL.csv", encoding='cp1252')
    df_column1 = ['Units Ordered']
    df_column2 = ['Ordered Product Sales']
    df_column3 = ['Units Ordered - B2B']
    df_column4 = ['Ordered Product Sales - B2B']
    df_total1 = column_total(df, df_column1).sum()
    df_total1 = df_total1.astype(float)
    df_total2 = column_total(df, df_column2).sum()
    df_total2 = df_total2.astype(float)
    df_total3 = column_total(df, df_column3).sum()
    df_total3 = df_total3.astype(float)
    df_total4 = column_total(df, df_column4).sum()
    df_total4 = df_total4.astype(float)
    cards = html.Div(
        [
            dbc.Row(
                [
                    dbc.Col(dbc.Card(generate_card_content("Total Sales", df_total2), color="success", inverse=True),md=dict(size=2,offset=3)),
                    dbc.Col(dbc.Card(generate_card_content("Total Orders", df_total1), color="warning", inverse=True),md=dict(size=2)),
                    dbc.Col(dbc.Card(generate_card_content("Total Sales B2B", df_total4), color="success", inverse=True),md=dict(size=2)),
                    dbc.Col(dbc.Card(generate_card_content("Total Orders B2B", df_total3), color="warning", inverse=True),md=dict(size=2))
                ],
                className="mb-4",
            ),
        ],id='card1'
    )
    return cards

In [11]:
df_norm= df.groupby('Class')[['Units Ordered', 'Sessions - Total', 'Page Views - Total', 'Ordered Product Sales']].sum()
df_norm["Class"] = df_norm.index
df_norm

Units Ordered  Sessions - Total  Page Views - Total  \
Class                                                                  
Decorative Pots            130              5011                7256   
Hanging Pots              2478             69337              102491   
Planters                  1287             36854               56228   
Tray                       292             11934               17594   
Water Can                   94              1908                2509   

                 Ordered Product Sales            Class  
Class                                                    
Decorative Pots                  51282  Decorative Pots  
Hanging Pots                   1313574     Hanging Pots  
Planters                        888172         Planters  
Tray                            174308             Tray  
Water Can                        36699        Water Can

In [12]:
df_b2b= df.groupby('Class')[['Units Ordered - B2B', 'Sessions – Total – B2B', 'Page Views – Total – B2B', 'Ordered Product Sales - B2B']].sum()
df_b2b["Class"] = df_b2b.index
df_b2b

Units Ordered - B2B  Sessions – Total – B2B  \
Class                                                          
Decorative Pots                    3                      24   
Hanging Pots                      57                     357   
Planters                          46                     303   
Tray                               7                      48   
Water Can                          0                      19   

                 Page Views – Total – B2B  Ordered Product Sales - B2B  \
Class                                                                    
Decorative Pots                        27                         1697   
Hanging Pots                          548                        31188   
Planters                              449                        33460   
Tray                                   68                         4493   
Water Can                              23                            0   

                           Class  
Class                             
Decorative Pots  Decorative Pots  
Hanging Pots        Hanging Pots  
Planters                Planters  
Tray                        Tray  
Water Can              Water Can

__DASH APP__

In [ ]:
# create Dash app
external_stylesheets = [dbc.themes.BOOTSTRAP]
app = dash.Dash(__name__, external_stylesheets=external_stylesheets)
app.title = 'Sales Dashboard'
colors = {'background': '#111111', 'bodyColor':'#F2DFCE', 'text': '#7FDBFF'}

# define the layout
app.layout = html.Div([
    html.H1(children='SALES DASHBOARD', style={'textAlign': 'center', 'color': colors['text'], 'backgroundColor': colors['background'],'font': {'size': 50}}), 
    html.Div(children='Sales Dashboard for the year 2022 (Jan-Dec 2022)', style={'textAlign':'center','color':colors['text'], 'backgroundColor': colors['background'], 'font': {'size': 40}}),
    generate_cards(),
    html.Div([html.Label('SELECT THE CUSTOMER TYPE')],id='my-div'+str(id),style={'textAlign':'center'}),
    dcc.Slider(
        id='customer_slider',
        min=0,
        max=1,
        marks={0: 'B2B', 1: 'Normal'}), 
    dcc.Graph(id='bar-chart1'),
    dcc.Graph(id='bar-chart2'),
    dcc.Graph(id='scatter-chart2'),
    dcc.Graph(id = 'scatter-chart3')
])

# define the callback function for the bar chart
@app.callback(
    Output(component_id='bar-chart1', component_property='figure'),
    Input(component_id='customer_slider',component_property= 'value')
)
def update_bar_chart1(customer_type):
    if customer_type == 1:
        fig = px.bar(df_norm, x='Class', y='Units Ordered', title=f'Class vs Units Ordered Bar Chart for Normal Customers')
        fig.update_layout(title_x=0.5,plot_bgcolor='#DADADA',paper_bgcolor='#A5D8DD')
    else :
        fig = px.bar(df_b2b, x='Class', y='Units Ordered - B2B', title=f' Class vs Units Ordered Bar Chart for B2B Customers')
        fig.update_layout(title_x=0.5,plot_bgcolor='#DADADA',paper_bgcolor='#A5D8DD')
    return fig

# define the callback function for the bar chart
@app.callback(
    Output(component_id='bar-chart2', component_property='figure'),
    Input(component_id='customer_slider', component_property='value')
)
def update_bar_chart2(customer_type):
    if customer_type == 1:
        fig = px.bar(df_norm, x='Class', y='Ordered Product Sales', title=f'Class vs Ordered Product Sales Bar Chart for Normal Customers')
        fig.update_layout(title_x=0.5,plot_bgcolor='#DADADA',paper_bgcolor='#A5D8DD')
    else:  
        fig = px.bar(df_b2b, x='Class', y='Ordered Product Sales - B2B' ,title=f'Class vs Ordered Product Sales Bar Chart for B2B Customers')
        fig.update_layout(title_x=0.5,plot_bgcolor='#DADADA',paper_bgcolor='#A5D8DD')
    return fig  

@app.callback(
    Output(component_id='scatter-chart2', component_property='figure'),
    Input(component_id='customer_slider', component_property='value')
)
def update_scatter_chart2( customer_type):
    if customer_type == 1:
        fig = px.scatter(df_norm, x='Ordered Product Sales', y='Page Views - Total',size='Sessions - Total', color='Class', title=f'Ordered Product Sales Scatter Plot for Normal Customers')
        fig.update_layout(title_x=0.5,plot_bgcolor='#DADADA',paper_bgcolor='#A5D8DD')
    else:  
        fig = px.scatter(df_b2b, x='Ordered Product Sales - B2B', y='Page Views – Total – B2B',size='Sessions – Total – B2B', color='Class', title=f'Ordered Product Sales Scatter Plot for B2B Customers')
        fig.update_layout(title_x=0.5,plot_bgcolor='#DADADA',paper_bgcolor='#A5D8DD')
    return fig  

@app.callback(
    Output(component_id='scatter-chart3', component_property='figure'),
    Input(component_id='customer_slider', component_property='value')
)
def update_scatter_chart3( customer_type):
    if customer_type == 1:
        fig = px.scatter(df_norm, x='Units Ordered', y='Page Views - Total',size='Sessions - Total', color='Class', title=f'Units Ordered Scatter plot for Normal Customers')
        fig.update_layout(title_x=0.5,plot_bgcolor='#DADADA',paper_bgcolor='#A5D8DD')
    else:  
        fig = px.scatter(df_b2b, x='Units Ordered - B2B', y='Page Views – Total – B2B',size='Sessions – Total – B2B', color='Class', title=f'Units Ordered Scatter Plot for B2B Customers')
        fig.update_layout(title_x=0.5,plot_bgcolor='#DADADA',paper_bgcolor='#A5D8DD')
    return fig  


if __name__ == '__main__':
    app.run_server(debug=True,use_reloader=False, port=8063)

Dash is running on http://127.0.0.1:8063/

Dash is running on http://127.0.0.1:8063/

 * Serving Flask app "__main__" (lazy loading)
 * Environment: production
   Use a production WSGI server instead.
 * Debug mode: on


__STATISTICS__

In [ ]:
import scipy.stats as sps
import statsmodels as sm, statsmodels.formula.api as smf
normality_test_shapiro1 = sps.shapiro(df['Units Ordered'])
normality_test_shapiro2 = sps.shapiro(df['Units Ordered - B2B'])
print('p-value for Units Ordered:', normality_test_shapiro1)
print('p-value for Units Ordered B2B:', normality_test_shapiro2)

# Since the data is not normally distributed, we carry out Spearman's correlation test.

correlation_test_spearman1 = sps.spearmanr(df['Units Ordered'], df['Sessions - Total'])
correlation_test_spearman1
correlation_test_spearman2 = sps.spearmanr(df['Units Ordered'], df['Page Views - Total'])
correlation_test_spearman2
print ('Spearman result1', correlation_test_spearman1)
print ('Spearman result2', correlation_test_spearman2)

In [ ]:
normality_test_shapiro1 = sps.shapiro(df['Ordered Product Sales'])
normality_test_shapiro2 = sps.shapiro(df['Ordered Product Sales - B2B'])
print('p-value for Ordered Product Sales:', normality_test_shapiro1)
print('p-value for Ordered Product Sales B2B:', normality_test_shapiro2)

# Since the data is not normally distributed, we carry out Spearman's correlation test.

correlation_test_spearman1 = sps.spearmanr(df['Ordered Product Sales'], df['Sessions - Total'])
correlation_test_spearman1
correlation_test_spearman2 = sps.spearmanr(df['Ordered Product Sales'], df['Page Views - Total'])
correlation_test_spearman2
print ('Spearman result1', correlation_test_spearman1)
print ('Spearman result2', correlation_test_spearman2)

In [ ]:
correlation_test_spearman1 = sps.spearmanr(df['Units Ordered - B2B'], df['Sessions – Total – B2B'])
correlation_test_spearman1
correlation_test_spearman2 = sps.spearmanr(df['Units Ordered - B2B'], df['Page Views – Total – B2B'])
correlation_test_spearman2
print ('Spearman result1', correlation_test_spearman1)
print ('Spearman result2', correlation_test_spearman2)

In [ ]:
correlation_test_spearman1 = sps.spearmanr(df['Ordered Product Sales - B2B'], df['Sessions - Total'])
correlation_test_spearman1
correlation_test_spearman2 = sps.spearmanr(df['Ordered Product Sales - B2B'], df['Page Views - Total'])
correlation_test_spearman2
print ('Spearman result1', correlation_test_spearman1)
print ('Spearman result2', correlation_test_spearman2)

## OBSERVATIONS

a. _NORMAL CUSTOMERS_:

 __1. Total Sales (Dashboard):__ Total sales from the Normal customers for the year 2022 was __rupees 24,64,035__.

 

__2. Total Orders (Dashboard):__ Total orders from the Normal customers for the year 2022 was __4,281__.

 

__3. Top 2 and Bottom 2 categories based on the sessions, page view, unit ordered, ordered product sales__: 

  **a. Based on Units ordered**

    Top 2 products: 
      1. Hanging Pots (2478)
      2. Planters (1287)

    Bottom 2 products:
      1. Water Can (94)
      2. Decorative Pots (130)

  **b. Based on Sessions Total**

    Top 2 Category: 
      1. Hanging Pots (69337)
      2. Planters (36854)

    Bottom 2 Category:
      1. Water Can (1908)
      2. Decorative Pots (5011)

  **c. Based on Page Views Total**

    Top 2 Category: 
      1. Hanging Pots (102491)
      2. Planters (56228)

    Bottom 2 Category:

      1. Water Can (2509)
      2. Decorative Pots (130)

 **d. Based on Ordered Products Sales**

    Top 2 Category: 

      1. Hanging Pots (1313574)
      2. Planters (888172)

    Bottom 2 Category:
      1. Water Can (36699)
      2. Decorative Pots (51282)


 


 **4. Relation between the page view/sessions with the units ordered and the ordered products sales**
 
 The reults from the Spearman's Correlation suggest that Units ordered and Sessions total have a strong positive correlation. The graphs indicate a positive relationship between units ordered of every category of the product and the sessions total. The units ordered is highest for the class which has the highest number of sessions total and the pattern is followed across all the classes. The units ordered increased with increase in the session total.
Further, the graphs indicate a positive relationship between ordered product sales of every category of the product and the sessions total. The ordered product sales increased with increase in the sessions total.
The same pattern was observed for page views total as well for both, units ordered and ordered product sales.


**5. STATISTICAL ANALYSIS**

*a. For Units Ordered*

Spearman result for Units Ordered and Sessions Total is (correlation=0.8533286459452503, pvalue=6.597723744012336e-27)
Spearman result for Units Ordered and Page Views Total is (correlation=0.8525244803975802, pvalue=8.261809544322544e-27)

The Spearman results for Units Ordered and both Sessions Total and Page Views Total indicate a strong positive correlation, with a very low p-value. This means that there is a statistically significant relationship between Units Ordered and both Sessions Total and Page Views Total, suggesting that as one of these variables increases, the other one is likely to increase as well.


*b. For Ordered Product Sales*

Spearman result for Ordered Product Sales and Sessions Total is(correlation=0.8356866965575063, pvalue=6.893123197322793e-25)
Spearman result for Ordered Product Sales and Page views Total is (correlation=0.8336552064845679, pvalue=1.136356753064204e-24)

The Spearman correlation between Ordered Product Sales and Sessions Total is 0.836, which indicates a strong positive correlation. This means that as the number of sessions total increases, the number of ordered products also increases. Furthermore, the associated p-value of 6.893e-25 indicates that this relationship is statistically significant.

The Spearman correlation between Ordered Product Sales and Page views Total is 0.662, which indicates a moderate positive correlation. This means that as the number of page views total increases, the number of ordered product total total also increases, though not as strongly as with sessions total. Furthermore, the associated p-value of 9.290e-13 indicates that this relationship is statistically significant.



 
**-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------**
 
 
 
 
*b. B2B CUSTOMERS:*


__1. Total Sales (Dashboard):__ Total sales from the B2B customers for the year 2022 was __rupees 70,838__.




__2. Total Orders (Dashboard):__ Total orders from the B2B customers for the year 2022 was __113__.



**3. Top 2 and Bottom 2 categories based on the sessions, page view, unit ordered, ordered product sales**

**a. Based on Units ordered**

    Top 2 products: 
      1. Hanging Pots (57)
      2. Planters (46)

    Bottom 2 products:
      1. Water Can (0)
      2. Decorative Pots (3)

**b. Based on Sessions Total**

    Top 2 products: 

      1. Hanging Pots (357)
      2. Planters (303)

    Bottom 2 products:
      1. Water Can (19)
      2. Decorative Pots (24)

**c. Based on Page Views Total**

    Top 2 products: 
    
      1. Hanging Pots (548)
      2. Planters (449)

    Bottom 2 products:
      1. Water Can (23)
      2. Decorative Pots (27)

**d. Based on Ordered Products Sales**

    Top 2 products: 

      1. Planters (33460)
      2. Hanging Pots (31188)

    Bottom 2 products:
      1. Water Can (0)
      2. Decorative Pots (1697)

**4. Relation between the page view/sessions total with the units ordered**

The graphs indicate a positive relationship between units ordered of every category of the product and the sessions total. the units ordered is highest for the class which has the highest number of sessions total and the pattern is followed across all the classes. The units ordered increased with increase in the session total.
Further, the graphs indicate a lpositive relationship between ordered product sales of every category of the product and the page views total. The ordered product sales increased with increase in the sessions total.
The same pattern was observed for page views total as well for both, units ordered and ordered product sales.

**5.  STATISTICAL ANALYSIS**

*a. For Units Ordered*

Spearman result for Units Ordered and Sessions Total is (correlation=0.6982112850299331, pvalue=1.4406884168470162e-14)
Spearman result for Units Ordered and Page Views Total is (correlation=0.7089527718267283, pvalue=3.746848670349896e-15)


The Spearman correlation results show that there is a strong positive correlation between Units Ordered and both Sessions Total and Page Views Total. This means that as the number of units ordered increases, the number of sessions and page views also increases. This indicates that people are likely to visit the website more and view more pages when they are more likely to purchase products. Additionally, the very small p-value for both correlations indicates that this relationship is statistically significant.


*b. For Ordered Product Sales*

Spearman result for Ordered Product Sales and Sessions Total is (correlation=0.6584105358734057, pvalue=1.3144418710477669e-12)
Spearman result for Ordered Product Sales and Page views Total is (correlation=0.6616983459185682, pvalue=9.29054441694643e-13)

The Spearman result for Ordered Product Sales and Sessions Total shows a moderate positive correlation of 0.6584 with a very low p-value of 1.314e-12. The Spearman result for Ordered Product Sales and Page views Total also shows a moderate positive correlation of 0.6617 with a very low p-value of 9.29e-13. This indicates that there is a strong positive relationship between Ordered Product Sales and both Sessions Total and Page views Total.


# MANAGERIAL IMPLICATIONS

**a. Normal Customers**

The above data suggests that Hanging Pots and Planters are the most popular and highest selling products among the Normal customers, while Water Can and Decorative Pots are the least popular and lowest selling products. 
This implies that the company should focus on increasing the sales of Hanging Pots and Planters by offering discounts, promotions, and other incentives. 

At the same time, the company should find ways to increase the sales of the least popular products, such as Water Can and Decorative Pots, by making them more attractive to customers. 
Additionally, the company should consider diversifying its product offerings to meet the changing needs of customers.
The company should consider analyzing customer data to identify customer preferences and tailor their marketing strategies accordingly. This will help them better understand their target audience and create products that meet their needs.

The data suggests that there is a positive relationship between the number of sessions and page views and the number of units ordered and ordered product sales. This implies that the more people visit the website and view the pages, the more likely they are to purchase products. Therefore, managers should focus on increasing the number of sessions and page views in order to drive sales. To do this, they should focus on marketing and promotional strategies to attract more visitors and viewers to the website. Additionally, managers should ensure that the content is engaging to encourage visitors to stay and explore the seller's page.


**b. B2B Customers**

The above data suggests that B2B customers are mostly interested in Hanging Pots and Planters, as they have the highest numbers in terms of units ordered, sessions, page views, and ordered product sales. 
This could indicate that these two categories should be given priority when it comes to marketing and promotions. On the other hand, Water Cans and Decorative Pots have the lowest numbers in all categories and may need to be given more attention in order to increase their sales. 

Managers should analyze the data further to determine the reasons behind the variation in sales and take necessary steps to improve the sales of these categories. 
Additionally, they could identify various strategies to make the B2B customers more interested in these categories.

The managerial implications from this analysis are that there is a clear positive relationship between units ordered of every category of the product and the page views total. This implies that managers should focus on increasing page views total in order to increase sales of their products. 

In addition, managers should pay attention to the different categories of the product and make sure that there is enough focus on each category to ensure that it is properly marketed and promoted in order to increase sales.
Increasing the visibility of the product in the market will lead to an increase in the units ordered and ordered product sales.


---**THE END**---